### ISSUE TITLE
lua entry thread aborted [attempt to index local 'upstream' (a nil value)]

### ISSUE THREAD

**Issue by vaxilicaihouxian** [Kong/kong#7655]
### Summary
- Kong version  2.2.2
- A concise summary of the bug.
All the request was broken and return http code 500.

When I set `worker_consistency` to `eventual` , after a while it happens

Here is the error.log:

```
2021/08/04 12:17:17 [error] 93#0: *517679 lua entry thread aborted: runtime error: /usr/local/share/lua/5.1/kong/runloop/balancer.lua:611: attempt to index local 'upstream' (a nil value)
stack traceback:
coroutine 0:
        /usr/local/share/lua/5.1/kong/runloop/balancer.lua: in function 'get_balancer'
        /usr/local/share/lua/5.1/kong/runloop/balancer.lua:996: in function 'execute'
        /usr/local/share/lua/5.1/kong/runloop/handler.lua:894: in function 'balancer_execute'
        /usr/local/share/lua/5.1/kong/runloop/handler.lua:1296: in function 'after'
        /usr/local/share/lua/5.1/kong/init.lua:755: in function 'access'
        access_by_lua(nginx-kong.conf:89):2: in main chunk, 
```

**Comment by locao** [comment]
Hello @vaxilicaihouxian! Thanks for your report. That doesn't seem like Kong 2.4, can you confirm what version you are using, please? I recommend you to upgrade to the most recent Kong version. There were several fixes to eventual worker consistency in Kong 2.4.

**Comment by vaxilicaihouxian** [comment]
> Hello @vaxilicaihouxian! Thanks for your report. That doesn't seem like Kong 2.4, can you confirm what version you are using, please? I recommend you to upgrade to the most recent Kong version. There were several fixes to eventual worker consistency in Kong 2.4.

Sorry about that.my version is 2.2.2
